scrabble-scorer.js

```diff
@@ -33,26 +33,71 @@ function oldScrabbleScorer(word) {
 // don't change the names or your program won't work as expected. //
 
 function initialPrompt() {
-   console.log("Let's play some scrabble! Enter a word:");
-};
+   let userInput = input.question("Let's play some scrabble!\n\nEnter a word: ");
+   return userInput;
+}
 
-let simpleScore;
+let simpleScore = {
+  name: 'Simple Score',
+  description: 'Each letter is worth 1 point.',
+  scoreFunction: function(wordToGrade) {return wordToGrade.length;}
+};
 
-let vowelBonusScore;
+let vowelBonusScore = {
+  name: 'Bonus Vowels',
+  description: 'Vowels are 3 pts, consonants are 1 pt.',
+  scoreFunction: function(wordToGrade) {
+    wordToGrade = wordToGrade.toUpperCase();
+    let score = 0;
+    const vowels = ['A','E','I','O','U'];
+    for(let i=0; i<wordToGrade.length; i++) {
+      if (vowels.indexOf(wordToGrade[i]) > -1) {
+        score += 3;
+      } else {
+        score++;
+      }
+    }
+    return score;
+  }
+};
 
-let scrabbleScore;
+let scrabbleScore = {
+  name: 'Scrabble',
+  description: 'The traditional scoring algorithm.',
+  scoreFunction: function(wordToGrade) {
+    wordToGrade = wordToGrade.toLowerCase();
+    let sum = 0;
+    for(let i=0; i<wordToGrade.length; i++) {
+      sum = sum + Number(newPointStructure[wordToGrade[i]]);
+    }
+    return sum;
+  }
+};
 
-const scoringAlgorithms = [];
+const scoringAlgorithms = [simpleScore, vowelBonusScore,scrabbleScore];
 
-function scorerPrompt() {}
+function scorerPrompt() {
+  let choice = input.question(`\n0 - Simple: One point per character\n1 - Vowel Bonus: Vowels are worth 3 points\n2 - Scrabble: Uses scrabble point system\n\nEnter 0, 1, or 2: `);
+  return Number(choice);
+}
 
-function transform() {};
+function transform(objectToChange) {
+  for (item in objectToChange) {
+    for (let i=0; i<objectToChange[item].length; i++) {
+      let letter = objectToChange[item][i];
+      letter = letter.toLowerCase();
+      newPointStructure[letter] = item;
+    }
+  }
```

Comment:
transform should return an object with the new key value pair.  key being a string and value being a number.